### PR TITLE
Allow decoding b64 string without header in processing utils

### PR DESCRIPTION
**Pull request by 1lint**
# Description
The user may pass a b64 string without a header (i.e. missing "data:image/png;base64,"), which can still be decoded, for example [stable-diffusion-webui api](https://github.com/AUTOMATIC1111/stable-diffusion-webui) expects payload with headers, but the [controlnet extension](https://github.com/Mikubill/sd-webui-controlnet) expects payload without headers. 

This change allows decoding the payload by using `image_encoded = encoding.rsplit(",", 1)[-1]`. The payload can have no commas, 1 comma, or multiple commas, this line will always select payload portion from the last comma to end of the string. 

I tested the change with 

```
from gradio_client.utils import decode_base64_to_binary
from gradio.processing_utils import decode_base64_to_image
from PIL import Image
from io import BytesIO

payload = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABgAAAAYCAYAAADgdz34AAAACXBIWXMAAAsTAAALEwEAmpwYAAABpklEQVR4nN2Uu0oEQRBFjyu+QDbyF0QQ/AnBQKNVZE38AvGBkbEowhoKamiiiyZiZmSi+AM+cMU1UWMDH7i6yEjDHSh6e2ZWQQMvNDNddbtud3V1wX9HO1AEykAFeNGoyFYU50cYA26BKGNUgdHvBM4Bq16AOaAP6AS6jO/c/Je0NhNx8Dowr0VHgZ3fyTcFvBmRzLQ44gcwZOx+8AdgxPgHgZp8haTg7SbnC54v0kjDlDg3QFuIUDSBJmQ79XaeBpeuM/HGQ4SyCVQXKTLjmGxMi7sdcl7LuWdEooydL3vzfvHdO2nAs5x5YLHJ1ETAOtCieVzCLlYDnuTs1nwp5XFZATc2zRtYThKIU+SOSYZIDGvbAlplnySAHRFnPPuS7I+BNb7wblKJ2jK9CDz5KOEuQqc7ADqSHlpVJFduFvcmwEmGgBuHuvAGjIpQ81rFsHpP0h186LtPEygZkVkvXSGBNTVF9/8J9GYJ5IxIBFxKaCAgsKJv3pT5Bk2ioMaVVqYWa/K9Aj3NirSpJ7necqUH9J7AdalxKXLl7Nr3r8CdOu4Ef4MvDU29eGwoPe8AAAAASUVORK5CYII="

payload_without_header = payload.split(",")[-1]

img1 = decode_base64_to_image(payload)

img2 = decode_base64_to_image(payload_without_header)

binary_payload, extension = decode_base64_to_binary(payload)
img3 = Image.open(BytesIO((binary_payload)))

binary_payload, extension = decode_base64_to_binary(payload_without_header)
img4 = Image.open(BytesIO((binary_payload)))

assert img1 == img2 == img3 == img4, "Images don't match"
```

In the process of making this change, I noticed the scripts/install_gradio.sh did not install gradio-client from the local source, so I added lines to install the package accordingly. I didn't want to touch the requirements.txt file since that would impact building, so gradio-client would be installed twice, once from the pip package uploaded to PyPI, and then from the local source in edittable mode. 

Minor changes so did not update CHANGELOG.md

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added a short summary of my change to the CHANGELOG.md
- [x] My code follows the style guidelines of this project
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by abidlabs** [comment]
Thanks @1lint for this PR! Looks like a reasonable change to me. Would you please add your test to `client\python\test\test_utils.py` so we don't break this in the future? 

(You can use `client\python\gradio_client\media_data.BASE64_IMAGE` as the base64 data)

**Comment by abidlabs** [comment]
Also, I do think this is worth adding to the changelog, so we can credit you properly!

File: scripts/install_gradio.sh [hunk 1]
```diff
@@ -6,4 +6,7 @@ source scripts/helpers.sh
 pip_required
 
 echo "Installing Gradio..."
-pip install -e .
\ No newline at end of file
+pip install -e .
+
+echo "Installing Gradio Client..."
+pip install -e client/python
```
Comment (abidlabs) [inline scripts/install_gradio.sh:12]: We use `scripts/install_gradio.sh` in a GH action where we don't want to install client/python from scratch. But that's okay, I'll just modify the GH action and push to this branch

Comment (1lint) [reply scripts/install_gradio.sh:12]: sounds good, thanks!

**Comment by 1lint** [comment]
Just updated the `client\python\test\test_utils.py`, adding on to the preexisting `test_decode_base64_to_binary` to check the decoded binaries remained the same

**Comment by abidlabs** [comment]
LGTM thanks @1lint for making the updates! Will merge once this runs through CI and all of the tests pass

**Review by abidlabs — APPROVED** [review]

**Comment by abidlabs** [comment]
Merging, thanks @1lint for the helpful PR!

**Comment by 1lint** [comment]
Great, thanks for the help getting this in! Cheers 😃

**Pull request merged**